#!/usr/bin/env python
PACKAGE = "cob_3d_experience_mapping"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("est_occ", int_t, 0, "Estimated occurences of unseen features",    10, 0,   10000)
gen.add("min_age", int_t, 0, "Min. age of a state for sensor input [number of states]",    10, 0,   1000)
gen.add("max_active_states", int_t, 0, "Maximum number of active states for optimization of execution time [number of states]",    1000, 0,   100000)
gen.add("deviation_factor", double_t, 0, "Variance of odometry [percentage]",    0.05, 0,   1)

proximity = gen.add_group("proximity")
proximity.add("translation", double_t, 0, "Translation distance between states [m]",    .5, 0,   10)
proximity.add("rotation", double_t, 0, "Rotation distance between states [rad]",    .5, 0,   10)

exit(gen.generate(PACKAGE, "experience_mapping", "Settings"))
